### ISSUE TITLE
No compile-time errors/warnings, but the program fails with TypeLoadException at runtime

### ISSUE THREAD

**Issue by VladimirReshetnikov** [dotnet/roslyn#23335]
**Version Used**: 

Version 15.5.0 Preview 4.0
VisualStudio.15.Preview/15.5.0-pre.4.0+27110.0

Visual C# 2017 00369-60000-00001-AA127
Microsoft Visual C# 2017

C# 7.2

**Steps to Reproduce**:

Compile and run:

```
struct S
{
    static S? X;

    static void Main() { }
}
``` 

**Actual Behavior**:
No compile-time errors or warnings, but the program fails at runtime:

> Unhandled Exception: System.TypeLoadException: Could not load type 'S' from assembly 'ConsoleApp1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.

**Expected Behavior**:
Earlier versions of the C# compiler used to issue the following compile-time error for this code:
> error CS0523: Struct member 'S.X' of type 'System.Nullable&lt;S&gt;' causes a cycle in the struct layout

The compiler should give the same error
**-OR-**
The runtime should be fixed to allow this kind of cyclic references (because it does not result in an unbounded number of generic types to be loaded).

**Comment by gafter** [comment]
Dup of https://github.com/dotnet/roslyn/issues/20624
Is actually a bug in core reference assembly
https://github.com/dotnet/corefx/issues/25251
